listen.go

```diff
@@ -26,20 +28,20 @@ func (r *Relay) Matches(a ma.Multiaddr) bool {
 }
 
 func (l *RelayListener) Accept() (tpt.Conn, error) {
-	ctx := l.Relay().ctx
 	select {
 	case c := <-l.incoming:
-		log.Infof("accepted relay connection: %s", c.ID())
-		s := RelayStatus{
-			Code:    StatusOK,
-			Message: "OK",
-		}
-		if err := s.WriteTo(c); err != nil {
+		err := l.Relay().writeResponse(c.Stream, pb.CircuitRelay_SUCCESS)
+		if err != nil {
+			log.Debugf("error writing relay response: %s", err.Error())
+			c.Stream.Close()
```

Comment:
The proxy is not in effect yet, so my comment about the transparent proxying is largely irrelevant.

You are right that we do want to close the output side of the channel in some way, but on the other hand the other side is not doing any writing at this point -- it is waiting for a response per protocol.
I will add a comment here.